godotengine/godot
# FileDialog open File and Dir MODE_OPEN_ANY

closes #4019

How to use/test?

Select a file and you will see a file selected print
Select a dir and you will see a dir selected print

```
extends Node2D

onready var filedialog = get_node( "FileDialog" )

func _ready():
    filedialog.connect("dir_selected", self,"_on_FileDialog_dir_selected")
    filedialog.connect("file_selected", self,"_on_FileDialog_file_selected")
    filedialog.connect("files_selected", self,"_on_FileDialog_files_selected")

    filedialog.set_mode( filedialog.MODE_OPEN_ANY )
    #filedialog.set_mode( filedialog.MODE_SAVE_FILE )
    #filedialog.set_mode( filedialog.MODE_OPEN_FILES )
    #filedialog.set_mode( filedialog.MODE_OPEN_FILE )
    filedialog.popup()

func _on_FileDialog_dir_selected( dir ):
    print("dir selected: " + dir )


func _on_FileDialog_file_selected( path ):
    print("file selected: " + path )

func _on_FileDialog_files_selected( paths ):
    print("files selected: " + str(paths) )
```
